dumbbells/dataset.py

```diff
@@ -0,0 +1,110 @@
+import gym
+import torch
+from torch.utils.data import Dataset as Ds, DataLoader
+
+
+class Dataset(Ds):
+    """Dataset containing information about the environment"""
+
+    def __init__(self, game, memory_size):
+        """Constructor, with the environment and memory_size specified
+
+        Args:
+            game: String containing the name of the environment to be created. Must be a valid environment. A list of
+                  valid environments can be found at "https://gym.openai.com/envs/#classic_control", or with
+                  gym.envs.registry.all()
+            memory_size: Max images we want to store at any time.
+
+        Returns: None
+        """
+
+        self.env = gym.make(game)
+        self.memory_size = memory_size
+        self.memory = []
+        self.position = 0
+
+        # For the environment "MountainCar-v0", there are 3 actions available:
+        # 0: Accelerate to the left
+        # 1: Don't accelerate
+        # 2: Accelerate to the right
+        self.action_space = self.env.action_space
+
+        # For the environment "MountainCar-v0", there are 2 observations, presented in the following order:
+        # Car Position: Ranges from -1.2 to 0.6
+        # Car Velocity: Ranges from -0.07 to 0.07
+        # The car starts at some position between -0.6 to -0.4
+        self.state_space = self.env.reset()
+
+        # For the environment "MountainCar-v0", there are 2 possible rewards:
+        # Reward of 0 is awarded if the agent has reached the flag (position 0.5)
+        # Reward of -1 is awarded if the agent is at a position < 0.5
+        self.reward_space = [-1, 0]
+
+    def get_state(self):
+        """Return the current state of the game (the observations)
+
+        Args: None
+
+        Returns:
+            Current state of the environment.
+        """
+        # For the environment "MountainCar-v0", this would return (Car_position, Car_velocity)
+        return self.state_space
+
+    def step(self, action):
+        """Take the given action in the environment and update the memory
+
+        Args:
+            action: Action the agent will take in the environment
+
+        Returns:
+            (State, Reward, Done, {}): The updated state, the reward, whether the game is done, and {}.
+        """
+        prev_state = self.get_state()
+        # Take the action
+        result = self.env.step(action)
+        # Update the state and memory
+        self.state_space = result[0]
+        self.pushMem(prev_state, action, result[1], self.get_state())
+        # Return the result of the action
+        return result
+
+    def pushMem(self, prev_state, action, reward, next_state):
```

Comment:
The pythonic variable naming standard uses all lowercase and `_`, so the name would be `push_mem` or `pushmem`, unless this is a convention form some library that I'm not aware of